pkg/apis/discovery/validation/validation.go

```diff
@@ -68,6 +72,7 @@ func ValidateEndpointSlice(endpointSlice *discovery.EndpointSlice) field.ErrorLi
 func ValidateEndpointSliceUpdate(newEndpointSlice, oldEndpointSlice *discovery.EndpointSlice) field.ErrorList {
 	allErrs := ValidateEndpointSlice(newEndpointSlice)
 
+	allErrs = append(allErrs, validateAddressType(newEndpointSlice.AddressType, supportedAddressTypes.Union(deprecatedAddressTypes))...)
```

Comment:
should not need this - if it can't change, you don't need to revalidate it?

Reply:
Good catch, thanks!